### ISSUE TITLE
About the implementation

### ISSUE THREAD

**Issue by wl082013** [liyunsheng13/BDL#38]
HI, Yunsheng, nice work. 
I was kind of confused when I was reading the paper. 
I understood the 'bi-directional' as simultaneous forward and backward pass', however when I check the code, I found it is  not simultaneous training. 

If I am correct, the training step  might be:
(1) train CycleGAN to get translated images
(2) train BDL.py to get segmentation model with the translated images and source images,
(3) train SSL.py to get pseudo-labels and refine the segmentation model.
(4) retrain CycleGAN with an additional perceptual loss. 

Then what's the next step? Step (1) tries to get better-translated images with the model of step (4)? 
And then repeat (2)-(3).  Does that mean this needs to be done multiple times based on the number of steps we want to try?
I am kind of confused about it since I thought 'bi-directional' was for simultaneous training of Im2Im translation and segmentation. 

Moreover, the CycleGAN folder is not complete. Some libraries are missing  :
from  util.image_pool import ImagePool
from .base_model import BaseModel
from fcn8s_LSD import FCN8s_LSD #lys
For the last one, fcn8s_LSD is the model from step (4)?

thanks for your help.

**Comment by liyunsheng13** [comment]
step (1) -> (4) can be iterated. You can just ignore from fcn8s_LSD import FCN8s_LSD, it is only for VGG net.
